packages/chrono-mongo-datastore/src/chrono-mongo-datastore.ts

```diff
@@ -114,6 +114,31 @@ export class ChronoMongoDatastore<TaskMapping extends TaskMappingBase>
     throw new Error(`Failed to insert ${String(input.kind)} document`);
   }
 
+  async delete<TaskKind extends keyof TaskMapping>(
+    taskId: string,
+  ): Promise<Task<TaskKind, TaskMapping[TaskKind]> | undefined> {
+    const task = await this.database
+      .collection<TaskDocument<TaskKind, TaskMapping[TaskKind]>>(this.config.collectionName)
+      .findOneAndDelete({
+        _id: new ObjectId(taskId),
+        status: TaskStatus.PENDING,
+      });
+
+    if (!task) {
+      const existingTask = await this.database
+        .collection<TaskDocument<TaskKind, TaskMapping[TaskKind]>>(this.config.collectionName)
+        .findOne({
+          _id: new ObjectId(taskId),
+        });
+
+      if (existingTask) {
+        throw new Error(`Task ${taskId} has a ${existingTask.status} status and can not be deleted.`);
+      }
+    }
+
+    return task ? this.toObject(task) : undefined;
```

Comment:
then we can also simplify this to
```suggestion
    return this.toObject(task);
```